spec/identity_spec.rb

```diff
@@ -16,24 +19,59 @@
     )
   end
 
-  before do
-    FileUtils.remove_dir(tmp_dir) if File.exists?(tmp_dir)
-    FileUtils.mkdir_p(tmp_dir)
-    File.write(File.join(tmp_dir, 'config'), identity_string)
+  before(:all) do
+    swarm_port=4061
+    api_port=5061
+    gateway_port=8090
+    @tmp_dir=ENV['TICTAC_TEST_DIR'] || File.join(__dir__, 'tmp')
+    @cfg=TicTac::Config.new(@tmp_dir)
+    puts @tmp_dir
+    config_file=JSON.load(File.open("#{@tmp_dir}/config"))
+    config_file["Addresses"]["Swarm"][0] = "/ip4/127.0.0.1/tcp/#{swarm_port}"
+    config_file["Addresses"]["API"] = "/ip4/127.0.0.1/tcp/#{api_port}"
+    config_file["Addresses"]["Gateway"] = "/ip4/127.0.0.1/tcp/#{gateway_port}"
+    File.write("#{@tmp_dir}/config",JSON.dump(config_file))
+    @external_ipfs=false
+    begin
+      ipfs_info=JSON.load(%x(ipfs -c "#{@tmp_dir}" id))
+      if ipfs_info["Addresses"] != nil
+        puts "external client detected"
+        @external_ipfs=true
+      end
+    rescue Exception => e
+      puts "no client detected"
+    end
+    if !@external_ipfs
+      @ipfs_proc=Open3.popen3("ipfs -c #{@tmp_dir} daemon 2>&1")
```

Comment:
ditto, and maybe this is just me but whenever I see `_proc` in a variable name I assume its callable, so it should respond to #call

Reply:
yeah we can call it 'ipfs_process'